magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/AspectPersistence.scala

```diff
@@ -13,27 +13,27 @@ import gnieh.diffson.sprayJson._
 import au.csiro.data61.magda.model.Registry._
 
 object AspectPersistence extends Protocols with DiffsonProtocol {
-  def getAll(implicit session: DBSession): List[AspectDefinition] = {
-    sql"select aspectId, name, jsonSchema from Aspects".map(rowToAspect).list.apply()
+  def getAll(implicit session: DBSession, tenantId: BigInt): List[AspectDefinition] = {
+    sql"select aspectId, name, jsonSchema, tenantId from Aspects where tenantId = $tenantId".map(rowToAspect).list.apply()
   }
 
-  def getById(implicit session: DBSession, id: String): Option[AspectDefinition] = {
-    sql"""select aspectId, name, jsonSchema from Aspects where aspectId=$id""".map(rowToAspect).single.apply()
+  def getById(implicit session: DBSession, id: String, tenantId: BigInt): Option[AspectDefinition] = {
+    sql"""select aspectId, name, jsonSchema, tenantId from Aspects where aspectId=$id and tenantId=$tenantId""".map(rowToAspect).single.apply()
   }
 
   def getByIds(implicit session: DBSession, ids: Iterable[String]): List[AspectDefinition] = {
     if (ids.isEmpty)
       List()
     else
-      sql"select aspectId, name, jsonSchema from Aspects where aspectId in ($ids)".map(rowToAspect).list.apply()
+      sql"select aspectId, name, jsonSchema, tenantId from Aspects where aspectId in ($ids)".map(rowToAspect).list.apply()
```

Comment:
How about mark it as TODO or create a github issue? I prefer marking as TODO.

Reply:
I don't mind making the bigger "how can we avoid this type of error in the future" part a github issue. But I think we need to fix all known cases of tenant data leakage before we merge this.

Reply:
Mike and I discussed offline and there are some issues. a=And leaking aspects is nearly a non-issue, so I'm ok with this being a separate github issue rather than addressed before merge.